utilities/general-helpers.metta

```diff
@@ -41,20 +41,20 @@
 ; The function uses isMember from general helpers functions.
 
 (: setDifference (-> Expression Expression Expression))
-(= (setDifference $set1 $set2) 
-    (if (== $set1 ()) () ; Base case: return empty set when $set1 is empty
-        (let* (
-            ($head (car-atom $set1))  ; Extract the head of set1
-            ($tail (cdr-atom $set1))  ; Extract the tail of set1
-            ($newtail (setDifference $tail $set2))  ; Recursively process the tail
-        )
-            ; Check if the $head is in set2, if so, skip it
-
-            (if (isMember $head $set2)
-                $newtail 
-                ; Otherwise, add the head to the new tail, avoiding unnecessary parentheses
-                (if (== (get-metatype $newtail) Expression) (cons-atom $head $newtail) $head)  ; Avoid wrapping the head if it's already properly structured
-            ))))
+(= (setDifference $set1 $set2) (collapse (subtraction (superpose $set1) (superpose $set2))))
```

Comment:
I understand the change you've made will improve the code, but we need the `setDifference` function to work in an ordered manner and we wanted to define our own function just incase the built in versions fail to preserve order from update to update (We've seen this happen). There are functions which depend on it so it's best to keep it as it is for the time being.